### PR TITLE
Fix floating number equality comparison in DeliveryCalculator

### PR DESCRIPTION
**Pull request by nexxome**
### 1. Why is this change necessary?
We had a case where a customer couldn't select a shipping method, because no existing price rule matched for his cart.
The shipping method has two price rules:

1. 0 € - 59,99 €
2. 60 € - ∞

screen3

The total cart amount was exactly 60,00 €, but the price rule didn't match in the DeliveryCalculator because floating numbers have limited precision in equality comparison (https://www.php.net/manual/en/language.types.float.php).

screen1

screen2

### 2. What does this change do, exactly?
This change adds a floating number equality check in `Shopware\Core\Checkout\Cart\Delivery\DeliveryCalculator::matches()`

### 3. Describe each step to reproduce the issue or behaviour.
It is not easy to create a shipping rule and cart to reproduce this issue, but I've added a test where the calculate method in DeliveryCalculator fails without the fix.

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Checkout/Cart/Delivery/DeliveryCalculator.php [hunk 1]
```diff
@@ -174,7 +174,7 @@ private function matches(Delivery $delivery, ShippingMethodPriceEntity $shipping
         }
 
         // $end (optional) exclusive
-        return ($value >= $start) && (!$end || $value <= $end);
+        return ($value >= $start || abs($value-$start) < 0.00001) && (!$end || $value <= $end || abs($value-$end) < 0.00001);
```
Comment (shyim) [inline src/Core/Checkout/Cart/Delivery/DeliveryCalculator.php]: Can you use this class?
https://github.com/shopware/platform/blob/trunk/src/Core/Framework/Util/FloatComparator.php#L5

**Comment by nexxome** [comment]
@shyim done

File: src/Core/Checkout/Cart/Delivery/DeliveryCalculator.php (hunk 1 above)
Comment (nexxome) [reply src/Core/Checkout/Cart/Delivery/DeliveryCalculator.php]: Done

**Comment by Dominik28111** [comment]
Thanks for your contribution 🎉. Your PR will be merged when our internal pipeline is green.

**Pull request closed without merging**